src/util/mod.rs

```diff
@@ -211,13 +211,7 @@ pub fn prepend_to_path(path: impl Display, base_path: impl Display) -> String {
 }
 
 pub fn command_present(name: &str) -> bossy::Result<bool> {
-    command_path(name).map(|_path| true).or_else(|err| {
-        if let Some(1) = err.status().and_then(|status| status.code()) {
-            Ok(false)
-        } else {
-            Err(err)
-        }
-    })
+    command_path(name).map(|output| output.status().success())
```

Comment:
This will only ever produce `Ok(true)`, since `map` will only be applied when the status was already a success. Likewise, this always produces `Err` when the status is non-zero.

Using the previous logic but with `if err.status().and_then(|status| status.code()).is_some()` as the condition instead of the previous `if let` should work. The `and_then` part isn't strictly necessary, but accounts for [an edge case](https://doc.rust-lang.org/std/process/struct.ExitStatus.html#method.code).

Reply:
Can you explain this better to me? I thought about that, but bossy's `run_and_wait_for_output` returns a `Result<Output>`. The call to `command_path` is basically a call to `something...run_and_wait_for_output`, so the output inside the Ok of this result is always a success? Did I get confused?

Reply:
I probably got confused about how bossy and `std::process::Command` worked didn't I?

Reply:
I've also just released bossy v0.2.1, which lets you just write `if err.code().is_some()` instead.

Reply:
Ah, sorry, I didn't see these responses!

Yeah, `bossy::Result` will only be `Ok` if the code is zero.

Reply:
I really feel embarrassed now. But thank you for the clarifications!